config/src/config/api_config.rs

```diff
@@ -13,43 +13,60 @@ use std::net::SocketAddr;
 #[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
 #[serde(default, deny_unknown_fields)]
 pub struct ApiConfig {
+    /// Enables the REST API endpoint
     #[serde(default = "default_enabled")]
     pub enabled: bool,
+    /// Address for the REST API to listen on.  Set to 0.0.0.0:<port> to allow all inbound connections.
     pub address: SocketAddr,
+    /// Path to a local TLS certificate to enable HTTPS
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub tls_cert_path: Option<String>,
+    /// Path to a local TLS key to enable HTTPS
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub tls_key_path: Option<String>,
-    // optional for compatible with old configuration
+    /// A maximum limit to the body of a POST request in bytes
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub content_length_limit: Option<u64>,
+    /// Enables failpoints for error testing
     #[serde(default = "default_disabled")]
     pub failpoints_enabled: bool,
+    /// Enables JSON output of APIs that support it
     #[serde(default = "default_enabled")]
     pub json_output_enabled: bool,
+    /// Enables BCS output of APIs that support it
     #[serde(default = "default_enabled")]
     pub bcs_output_enabled: bool,
+    /// Enables encode submission API
     #[serde(default = "default_enabled")]
     pub encode_submission_enabled: bool,
+    /// Enables transaction submission APIs
     #[serde(default = "default_enabled")]
     pub transaction_submission_enabled: bool,
+    /// Enables transaction simulation
     #[serde(default = "default_enabled")]
     pub transaction_simulation_enabled: bool,
-
+    /// Maximum number of transactions that can be sent with the Batch submit API
     pub max_submit_transaction_batch_size: usize,
-
-    // Maximum page size for paginated APIs
+    /// Maximum page size for transaction paginated APIs
     pub max_transactions_page_size: u16,
+    /// Maximum page size for event paginated APIs
     pub max_events_page_size: u16,
+    /// Maximum page size for resource paginated APIs
     pub max_account_resources_page_size: u16,
+    /// Maximum page size for module paginated APIs
     pub max_account_modules_page_size: u16,
-
-    /// Max gas unit for view function.
+    /// Maximum gas unit limit for view functions
```

Comment:
> Some high-level questions/concerns:
> 
> 1. What's the doc goal here? Is this to benefit internal devs, external devs, or users?
>    
>    * If it's users (e.g., node operators), I'd prefer we document them elsewhere (e.g., on dedicated component docs, like I have for state sync: https://aptos.dev/guides/state-sync). Users won't read these (and likely shouldn't).
>    * If it's internal (or external) devs, I'd argue that many of the comments are redundant (and devs shouldn't expect all code to be annotated -- this seems like an anti-pattern). Likewise, if we're going to do this, why not do it for all configs (instead of just this subset)?
> 2. Generally, I'm not a big fan of requiring comments on all struct fields, etc., and it's likely why we've never enforced the "missing docs" [lint](https://doc.rust-lang.org/rustdoc/lints.html#missing_docs), because it tends to result in really meaningless comments that tend to create baggage.
> 
> Also, you could probably unblock the optimization part of the PR by splitting the PR into two: (i) single node optimizations; and (ii) doc comments.
> 
> Interested to see what others think though... (I know I'm pretty opinionated sometimes 😄)
> 
> (Rest of the review is just nitty nits that can be taken or left)

I would argue, that these docs provide value in that someone can easily tell what a configuration is supposed to do.

I spent a lot of time fumbling through this configuration trying to figure out what "X" does.  In all of the state sync docs, there actually are comments on every single field, but aren't made into doc comments.

I'm not viewing that there should be a comment on "every struct", but I would argue something as key as configuration for the node should be easy to view as a reference, given that any docs in aptos.dev will lag behind these updated configurations.